lib/fog/vsphere/models/compute/volume.rb

```diff
@@ -49,7 +49,8 @@ def save
           requires :server_id, :size, :datastore
 
           if unit_number.nil?
-            used_unit_numbers = server.volumes.map { |volume| volume.unit_number }
+            # Vsphere maps unit_numbers 7 and greater to a higher SCSI ID since the pvscsi driver reserves SCSI ID 7
+            used_unit_numbers = server.volumes.map { |volume| volume.unit_number < 7 ? volume.unit_number : volume.unit_number - 1 }
```

Comment:
I can be just an inline comment as you did but instead of being in two parts it could be at the top level of the method explaining how it works and why this workaround is in place.